src/logic/proposals/encoding/encoding.ts

```diff
@@ -160,14 +160,21 @@ export const goEncodeEvmScript = async (
   // Build the EVM script according to the scheme
   const goBuildEvmScript = goSync(() => {
     // Build the call data that the EVMScript will use (and remove 0x prefix)
-    const callData =
-      encodedExecuteSignature +
-      utils.defaultAbiCoder
-        .encode(
-          ['address', 'uint256', 'bytes'],
-          [targetAddress, targetValue, encodeFunctionSignature(targetSignature) + encodedTargetParameters.substring(2)]
-        )
-        .substring(2);
+    const targetCallData =
+      // If this is a version 1 proposal, then we need to keep encoding the target signature regardless if it's blank
+      // or not, otherwise we could get a script mismatch and a false-positive for the malicious proposal check
+      version === '1' || targetSignature
```

Comment:
fwiw, I was thinking whether this should be moved to a separate function to better see the versioning complexity, but doing that is a bit random/premature (in the future the next version may alter different part of the code). So I think this fine.